ivy/stateful/helpers.py

```diff
@@ -301,7 +302,9 @@ def _compute_module_dict(self):
         self._module_dict = Container()
         for key, value in self.__dict__.items():
             if isinstance(value, ivy.Module):
-                if "stateful" in value.__module__:
+                if key in self._module_dict:
```

Comment:
we don't need this `if` check either right? Given that `self._module_dict` is empty before iterating and that the `key` would be present in the `self.__dict__` only once?